server/post.go

```diff
@@ -0,0 +1,115 @@
+package main
+
+import (
+	"fmt"
+
+	"github.com/mattermost/mattermost-plugin-msteams-meetings/server/remote"
+	"github.com/mattermost/mattermost-plugin-msteams-meetings/server/store"
+	"github.com/mattermost/mattermost-server/v5/model"
+	"github.com/pkg/errors"
+	msgraph "github.com/yaegashi/msgraph.go/beta"
+)
+
+func (p *Plugin) postMeeting(creator *model.User, channelID string, topic string) (*model.Post, *msgraph.OnlineMeeting, error) {
+	conf, err := p.getOAuthConfig()
+	if err != nil {
+		return nil, nil, err
+	}
+	userInfo, err := p.store.GetUserInfo(creator.Id)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	attendees := []*store.UserInfo{}
+
+	channel, appErr := p.API.GetChannel(channelID)
+	if appErr != nil {
+		return nil, nil, err
+	}
+
+	if channel.IsGroupOrDirect() {
+		var members *model.ChannelMembers
+		members, appErr = p.API.GetChannelMembers(channelID, 0, 100)
+		if appErr != nil {
+			return nil, nil, err
+		}
+		if members == nil {
+			return nil, nil, errors.New("returned members is nil")
+		}
+		for _, member := range *members {
+			var attendeeInfo *store.UserInfo
+			attendeeInfo, err = p.store.GetUserInfo(member.UserId)
+			if err != nil {
+				continue
+			}
+			attendees = append(attendees, attendeeInfo)
+		}
+	}
+
+	client := remote.NewClient(conf, userInfo.OAuthToken, p.API)
+
+	meeting, err := client.CreateMeeting(userInfo, attendees...)
```

Comment:
just curious why did you use variadic func?

Reply:
I guess I did originally to keep any `CreateMeeting(creator)` keep working after adding the changes, but I see that I only use this with a list of attendees. That being said, what do you think is best?